### PR TITLE
Overtrain and mode collapse detection

### PR DESCRIPTION
**Pull request by KawaiianPizza**
# GUI Changes

### Model Inference
- Inference page no longer auto-loads a model to save on VRAM

### Train
- Selecting a dataset folder from the dropdown menu sets the "Model Name" to the dataset name
- Added a checkbox for model collapse retraining
- Added a checkbox for overtrain detection
- Changed the default values for:
  - Training epochs
  - Save latest .ckpt
  - Save weights every save point
  - Reload on collapse
  - Stop on overtraining

# Other Changes

### Training
- When overtraining detection is enabled, the model will be trained for 100 epochs after the last best epoch was found
- When overtraining detection is enabled, the Tensorboard log interval will be set to every epoch instead of the default 200 steps and only update when a new best has been found
- When overtraining detection is enabled, the "eval" Tensorboard will be updated every epoch, not just the best ones.
- When reload on collapse is enabled, every epoch will be compared to the epoch before it. If the current epoch is 25% or less than the last epoch, it will stop training and go back to the last checkpoint and retrain from there.

### Bug Fixes
- When scanning for datasets, it now checks if it is a folder instead of checking if it contains a "." allowing datasets with names like "D.Va"

### CODE REVIEW

File: lib/infer/modules/train/train.py [hunk 1]
```diff
@@ -697,8 +702,139 @@ def train_and_evaluate(
...
+                )
+            if not continued:
+                logger.info(
+                    f"New best epoch!! [e{epoch}]\n"
+                    if epoch - best["epoch"] == 0
+                    else f'Last best epoch [e{best["epoch"]}] seen {epoch - best["epoch"]} epochs ago\n'
+                )
+            if continued:
+                continued = False
+            # if overtraining is detected, exit (idk what the 2333333 stands for but it seems like success ¯\_(ツ)_/¯)
```
Comment (HyperChocolat) [inline lib/infer/modules/train/train.py:823]: Man i love this line

**Comment by HyperChocolat** [comment]
Wow, nice pr. now we need to fix small things like putting ñ one the dataset or model name crashing applio
Also i have some questions. why did you nuked `value="rmvpe",` in [infer-web 2559](https://github.com/IAHispano/Applio-RVC-Fork/pull/104/files#diff-c09ed1cf9ba0cc622604384b041d27e57bf195a8cac45dbd28c1310d33ea1547L2559) or why change 750 to 999 in [infer-web 2614](https://github.com/IAHispano/Applio-RVC-Fork/pull/104/files#diff-c09ed1cf9ba0cc622604384b041d27e57bf195a8cac45dbd28c1310d33ea1547L2614) Putting that aside this pr probably need to delete go-tensorboard.bat cuz the link will be obtained in the webui anyways, right? (honestly, I didn't read everything but giving it a quick pass it seems to do it) thanks for your work!

**Comment by KawaiianPizza** [comment]
>  why did you nuked `value="rmvpe",` in [infer-web 2559](https://github.com/IAHispano/Applio-RVC-Fork/pull/104/files#diff-c09ed1cf9ba0cc622604384b041d27e57bf195a8cac45dbd28c1310d33ea1547L2559) or why change 750 to 999 in [infer-web 2614](https://github.com/IAHispano/Applio-RVC-Fork/pull/104/files#diff-c09ed1cf9ba0cc622604384b041d27e57bf195a8cac45dbd28c1310d33ea1547L2614) 
I forgot to change those back to default before committing. Those were the settings I used when checking for bugs in train.py and was a small optimization for that case, so it can be reverted as it's personal preference.

> Putting that aside this pr probably need to delete go-tensorboard.bat cuz the link will be obtained in the webui anyways, right?
I made tensorboard run on a different port so you can use go-tensorboard.bat at the same time if you want to look at the entire logs folder. With the changes, it only looks at the model currently being trained.

**Comment by HyperChocolat** [comment]
> I forgot to change those back to default before committing. Those were the settings I used when checking for bugs in train.py and was a small optimization for that case, so it can be reverted as it's personal preference.

Oh, ok then

> I made tensorboard run on a different port so you can use go-tensorboard.bat at the same time if you want to look at the entire logs folder. With the changes, it only looks at the model currently being trained.

Woa is even better than i thought then. well now is just wait to blaise, aitronssesin, TheUserNil 
to check this

**Review by blaise-tk — APPROVED** [review]

**Pull request merged**